### PR TITLE
soc: arm: nrf52810: reduce number of boot-time configured MPU regions

### LINKED ISSUE #13949: tests: Ztest problem - not booting properly
**Describe the bug**
Problem with ztest - not booting properly.

**To Reproduce**
Steps to reproduce the behavior (using samples/testing/integration):
1. mkdir build; cd build
2. cmake -DBOARD=nrf52810_pca10040
3. make flash

During test I was using RTT_console.

**Expected behavior**
Messages about successfully finished tests should be displayed.

**Screenshots or console output**
```
***** Booting Zephyr OS v1.14.0-rc1-624-ge731bdc81a3b *****
ASSERTION FAIL [num_parts <= max_partitions] @ /home/mapi/ncs/zephyr_original/zephyr/kernel/mem_domain.c:90
	
***** Kernel Panic! *****
Current thread ID = 0x20000748
Faulting instruction address = 0x1340

```

**Environment:**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.9.5
 - Commit: e731bdc81a3b2e2fdf498bccab9627b18fa38de6

**Additional context**
It used to work properly before. Sample commit when working: df7162331f831ac6a0ce108d7c512573ffb51ce0

### PR DESCRIPTION
**Pull request by ioannisg**
Fixes #13949

nRF52810 Flash and SRAM sizes do not have power-of-two values.
This commit reduces the amount of MPU regions reserved at boot
time to enforce basic permissions in Flash and SRAM memory. It
does this by selecting the next power-of-two size for the Flash
and SRAM regions, respectively. In this way we can cover the
entire Flash and SRAM space each with a single MPU region, thus,
save two MPU regions for use by the kernel/application.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
I believe it is safe to do this way, because the areas beyond the flash and SRAM boundaries are, anyways, covered by the background map (if MPU is enabled). The only functional difference is that now we even block execution fro the "non-existing" Flash and SRAM areas :) In any case, the additional memory ranges we cover inside the extended Flash and SRAM MPU regions are not implemented on the SOC, and should, anyways, cause BusFaults...

File: soc/arm/nordic_nrf/nrf52/mpu_mem_cfg.h [hunk 1]
```diff
@@ -14,9 +14,8 @@
 #elif CONFIG_FLASH_SIZE == 128
 #define REGION_FLASH_SIZE REGION_128K
 #elif CONFIG_FLASH_SIZE == 192
-#define REGION_FLASH_SIZE    REGION_128K
-#define REGION_FLASH_1_SIZE  REGION_64K
-#define REGION_FLASH_1_START CONFIG_FLASH_BASE_ADDRESS + KB(128)
+/* Use the next power-of-two size, to spare one MPU region. */
```
Comment (carlescufi) [inline soc/arm/nordic_nrf/nrf52/mpu_mem_cfg.h]: s/spare/save

File: soc/arm/nordic_nrf/nrf52/mpu_mem_cfg.h [hunk 2]
```diff
@@ -31,9 +30,8 @@
 
 /* SRAM Region Definitions */
 #if CONFIG_SRAM_SIZE == 24
-#define REGION_SRAM_0_SIZE  REGION_16K
-#define REGION_SRAM_1_SIZE  REGION_8K
-#define REGION_SRAM_1_START CONFIG_SRAM_BASE_ADDRESS + KB(16)
+#define REGION_SRAM_0_SIZE  REGION_32K
+/* Use the next power-of-two size, to spare one MPU region. */
```
Comment (carlescufi) [inline soc/arm/nordic_nrf/nrf52/mpu_mem_cfg.h]: s/spare/save

**Review by carlescufi — CHANGES_REQUESTED** [review]
Looks good, just doc

**Review by ioannisg — COMMENTED** [review]
Comments addressed, @carlescufi

**Review by tyrantlizard77 — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**